lib/unison-prelude/src/Unison/Util/Set.hs

```diff
@@ -9,6 +10,13 @@ import qualified Data.Maybe as Maybe
 import Data.Set (Set)
 import qualified Data.Set as Set
 
+-- | Set difference, but return @Nothing@ if the difference is empty.
+difference1 :: Ord a => Set a -> Set a -> Maybe (Set a)
```

Comment:
I know it can be a pain to mix Set types sometimes, but this seems like a good place to return a NonEmpty set since we've already done the "proof"

Reply:
Yeah, I think NESet is a reasonable thing to return here, if we want to incur that dependency in the prelude package, which seems fine

Reply:
Yeah if we already depend on a library in the project I don't really care which packages in the project depend on it.

Reply:
I don't think that's right, it's important to keep dependencies slim, especially for the prelude package which is meant to be imported by every other package.

Yeah you build it all together with a top-level `stack build`, but a properly modularized codebase will not have to all be built at once.

Reply:
Sure, the initial build might get a teensy amount more parallelism, but after that initial build all of the external deps should be cached anyways so I would suspect any effect would be minimal to trivial. 

Am I missing something that makes this a blocker?

Reply:
There's no blocker, I was responding to this point:

> Yeah if we already depend on a library in the project I don't really care which packages in the project depend on it, it all gets built together anyways.